docs/source/development/contributing/code_testing.md

```diff
@@ -0,0 +1,73 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.2
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# Package testing and profiling
+
+The `pyrealm` package uses `pytest` to provide benchmark tests, unit tests and
+integration testing. In addition, `doctest` is used to maintain examples of code usage
+in the package docstrings and ensure that the documented return values are correct.
+
+## Using `pytest`
+
+The `tests` directory contains modules providing test suites for each of the different
+package modules. This includes:
+
+* regression testing the output of `pyrealm` code against previously existing
+  implementations of some functionality, such as the `rpmodel` and `SPLASH` packages.
+* unit testing of individual functions and methods and,
+* profiling and integration testing using combinations of modules.
+
+These are the main tests that ensure that the package is behaving as expected and that
+it produces stable outputs. The test suite can be run from repository using:
+
+```bash
+poetry run pytest
+```
+
+The `pyproject.toml` file contains `pytest` configuration details.
+
+## Using `doctest`
+
+Some of the package docstrings contain `doctest` examples of code use. These examples
+are intended to provide simple examples of method or function use and generate an
+output: the `doctest` module is used to make sure that the code runs and gives the
+expected result.
+
+We have configured `pytest` to automatically also run `doctest`, but you can manually
+check the tests in files using, for example:
+
+```bash
+poetry run python -m doctest pyrealm/pmodel/pmodel.py
```

Comment:
Are we currently using doctest anywhere?

Reply:
We are now 😉